mesonbuild/linkers/detect.py

```diff
@@ -219,6 +219,19 @@ def guess_nix_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         linker = linkers.AIXDynamicLinker(
             compiler, for_machine, comp_class.LINKER_PREFIX, override,
             version=search_version(e))
+    elif 'IEW5033' in e:
+        # On z/OS the binder does not have a version flag. Instead, the binder
+        # version is tied to the OS version.
+        # For compatibility reasons `uname()` does not return the z/OS version
+        # by default.
+        p, o, e = Popen_safe(['/bin/uname', '-Irv'])
+        if p.returncode != 0:
+            __failed_to_detect_linker(compiler, check_args, o, e)
+        os_r, os_v = o.strip().split(' ')
+        version = f"{os_v.strip('0')}.{os_r.split('.')[0].strip('0')}"
```

Comment:
Please avoid too overly-complex f-strings in favor of e.g. temporary variables, as it's much harder to read complex code sections embedded inside an f-string.